src/main/java/io/debezium/connector/yugabytedb/YugabyteDBTablePoller.java

```diff
@@ -87,69 +92,142 @@ public void run() {
-          Set<TableInfo> responseSet = new HashSet<>(tableInfoSetFromResponse);
+        if (retryCount > MAX_RETRY_COUNT) {
+          LOGGER.error("Retries exceeded the maximum retry count in table poller thread,"
+                    + " all {} retries failed", MAX_RETRY_COUNT);
+          throw fail(e);
+        }
 
-          Set<TableInfo> intersection = new HashSet<>(cachedSet);
-          intersection.retainAll(responseSet);
+        LOGGER.warn("Exception while trying to get DB stream Info in poller thread,"
+                  + "will retry again", e);
+        return false;
+      }
+    }
 
-          Set<TableInfo> difference = new HashSet<>(responseSet);
-          difference.removeAll(cachedSet);
-          
-          if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Common tables between the cached table info set and the set received "
-                        + "from GetDBStreamInfoResponse:");
-            intersection.forEach(tableInfo -> {
-              LOGGER.debug(tableInfo.getTableId().toStringUtf8());
-            });
-
-            LOGGER.debug("New tables as received in the GetDBStreamInfoResponse: ");
-            difference.forEach(tableInfo -> {
-              LOGGER.debug(tableInfo.getTableId().toStringUtf8());
-            });
-          }
+    return false;
+  }
+
+  private boolean areThereNewTablesInPublication() {
+    // Reset the retry counter.
+    short retryCount = 0;
+
+    while (retryCount <= MAX_RETRY_COUNT) {
+      try {
+        boolean shouldRestart = false;
+        Set<String> tablesInPublication = getTablesInPublication();
 
-          for (TableInfo tableInfo : difference) {
-            if (isTableIncludedForStreaming(tableInfo.getTableId().toStringUtf8())) {
-              String message = "Found {} new table(s), signalling context reconfiguration";
-              LOGGER.info(message, difference.size());
-              shouldRestart = true;
+
+        if (cachedTableNameSet == null) {
+          LOGGER.debug("Cached table list in the poller thread is null, initializing it now");
+          cachedTableNameSet = tablesInPublication;
+        } else {
+          if (cachedTableNameSet.size() != tablesInPublication.size()) {
+            Set<String> cachedSet = new HashSet<>(cachedTableNameSet);
+            Set<String> responseSet = new HashSet<>(tablesInPublication);
+
+            Set<String> intersection = new HashSet<>(cachedSet);
+            intersection.retainAll(responseSet);
+
+            Set<String> difference = new HashSet<>(responseSet);
+            difference.removeAll(cachedSet);
+
+            if (LOGGER.isDebugEnabled()) {
+              LOGGER.debug("Common tables between the cached table info set and the set received "
+                          + "from GetDBStreamInfoResponse:");
+              intersection.forEach(table -> LOGGER.debug(table));
+
+              LOGGER.debug("New tables as received in the GetDBStreamInfoResponse: ");
```

Comment:
Wrong log messages here and above. We are not using `GetDBStreamInfoResponse` here.